Lucid/Stores/RemoteStore.swift

```diff
@@ -51,14 +53,24 @@ public final class RemoteStore<E>: StoringConvertible where E: RemoteEntity {
     // MARK: - API
 
     public func get(withQuery query: Query<E>, in context: ReadContext<E>, completion: @escaping (Result<QueryResult<E>, StoreError>) -> Void) {
-
         Task {
             let result = await self.get(withQuery: query, in: context)
             completion(result)
         }
     }
 
     public func get(withQuery query: Query<E>, in context: ReadContext<E>) async -> Result<QueryResult<E>, StoreError> {
+        do {
+            return try await processingQueue.enqueueBarrier { completion in
```

Comment:
Why do we need processing barriers here?

Reply:
I think this is mainly for the Combine -> Async/Await logic. I was seeing tests that were behaving non-deterministically and the operations performing out of order.

Since this logic is really just about ordering onto the client queue, I think the wait time is minimal.

Reply:
Got it, I was thinking the barrier would prevent concurrent operations but this is RemoteStore not ClientQueue 😵‍💫

Reply:
This is the test the led me to make the change:
```swift
    func test_two_different_searches_should_send_two_requests_to_the_client_using_endpoint_derived_from_entity_type_async() async {

        let requestConfig = APIRequestConfig(method: .get, path: .path("fake_entity"))
        clientQueueSpy.responseStubs[requestConfig] = APIClientQueueResult<Data, APIError>.success(APIClientResponse(data: payloadStubData, cachedResponse: false))

        await withTaskGroup(of: Void.self) { group in
            group.addTask {
                let result = await self.store.search(withQuery: .filter(.title == .string("fake_title")), in: self.derivedFromEntityTypeContext)
                switch result {
                case .success(let entities):
                    XCTAssertEqual(entities.isEmpty, false)
                case .failure(let error):
                    XCTFail("Unexpected error: \(error)")
                }
            }

            group.addTask {
                let result = await self.store.search(withQuery: .filter(.identifier == .identifier(EntitySpyIdentifier(value: .remote(24, nil)))), in: self.derivedFromEntityTypeContext)
                switch result {
                case .success(let entities):
                    XCTAssertEqual(entities.isEmpty, false)
                case .failure(let error):
                    XCTFail("Unexpected error: \(error)")
                }
            }
        }

        XCTAssertEqual(self.clientQueueSpy.appendInvocations.count, 1)
    }
```

Previously the check for `appendInvocations` would always be 1, which indicates that it would process one task operation at a time. Now it sometimes gives us a value of `2` (sometimes still `1`).

This seems to indicate that previously `Task` blocks were being a bit more ordered/sequential, and now they are more overlapping/concurrent. So far the performance in the app seems as good as ever, so I don't think this should introduce an issue.